Rakefile

```diff
@@ -1,12 +1,29 @@
 require 'rake'
 require 'rspec/core/rake_task'
 require 'puppetlabs_spec_helper/rake_tasks'
+require 'puppet'
+
 begin
   require 'beaker/tasks/test' unless RUBY_PLATFORM =~ /win32/
 rescue LoadError
   #Do nothing, only installed with system_tests group 
 end
 
+
+# If puppet does not support symlinks (e.g., puppet <= 3.5) we cannot use
+# puppetlabs_spec_helper's `rake spec` task because it requires symlink
+# support. Redefine `rake spec` to avoid calling `rake spec_prep` (requires
+# symlinks to place fixtures) and restrict the pattern match only files under
+# the 'unit' directory (tests in other dirs require fixtures).
+if Puppet::Util::Platform.windows? and !Puppet::FileSystem.respond_to?(:symlink)
+  ENV["SPEC"] = "./spec/unit/**/*_spec.rb"
```

Comment:
Chocolatey module doesn't use that folder, but others do, so I'm thinking of a more reusable solution.